nifi-commons/nifi-flow-encryptor/src/main/java/org/apache/nifi/flow/encryptor/JsonFlowEncryptor.java

```diff
@@ -0,0 +1,118 @@
+        final JsonFactory factory = new JsonFactory();
+        try (final JsonGenerator generator = factory.createGenerator(outputStream)){
+            try (final JsonParser parser = factory.createParser(inputStream)) {
+                parser.setCodec(new ObjectMapper());
+                processJsonByTokens(parser, generator, inputEncryptor, outputEncryptor);
+            }
+        } catch (IOException e) {
+            throw new UncheckedIOException("Failed Processing Flow Configuration", e);
+        }
+    }
+
+    private void processJsonByTokens(final JsonParser parser, final JsonGenerator generator,
+                                     final PropertyEncryptor inputEncryptor, final PropertyEncryptor outputEncryptor) throws IOException {
+        while (true) {
+            final JsonToken token = parser.nextToken();
+            if (token == null) {
+                return;
+            }
+
+            switch (token) {
+                case NOT_AVAILABLE:
+                    break;
+                case START_OBJECT:
+                    generator.writeStartObject();
+                    break;
+                case END_OBJECT:
+                    generator.writeEndObject();
+                    break;
+                case START_ARRAY:
+                    generator.writeStartArray();
+                    break;
+                case END_ARRAY:
+                    generator.writeEndArray();
+                    break;
+                case FIELD_NAME:
+                    generator.writeFieldName(parser.getValueAsString());
+                    break;
+                case VALUE_EMBEDDED_OBJECT:
+                    generator.writeEmbeddedObject(parser.getEmbeddedObject());
+                    break;
+                case VALUE_STRING:
+                    final String value = parser.getValueAsString();
+                    final Matcher matcher = ENCRYPTED_PATTERN.matcher(value);
+                    if (matcher.matches()) {
+                        generator.writeString(getOutputEncrypted(matcher.group(FIRST_GROUP), inputEncryptor, outputEncryptor));
+                    } else {
+                        generator.writeString(value);
+                    }
+                    break;
+                case VALUE_NUMBER_INT:
+                    generator.writeNumber(parser.getIntValue());
+                    break;
+                case VALUE_NUMBER_FLOAT:
+                    generator.writeNumber(parser.getFloatValue());
+                    break;
+                case VALUE_TRUE:
+                    generator.writeBoolean(true);
+                    break;
+                case VALUE_FALSE:
+                    generator.writeBoolean(false);
+                    break;
+                case VALUE_NULL:
+                    generator.writeNull();
+                    break;
+                default:
+                    break;
```

Comment:
It seems better to through an `IllegalStateException()` here to ensure the method covers all JSON token types.